go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -503,10 +503,11 @@ func delServiceRules(service *kapi.Service, npw *nodePortWatcher) {
 	delGatewayIptRules(service, false)
 }
 
-func serviceUpdateNeeded(old, new *kapi.Service) bool {
+func serviceUpdateNotNeeded(old, new *kapi.Service) bool {
 	return reflect.DeepEqual(new.Spec.Ports, old.Spec.Ports) &&
 		reflect.DeepEqual(new.Spec.ExternalIPs, old.Spec.ExternalIPs) &&
 		reflect.DeepEqual(new.Spec.ClusterIP, old.Spec.ClusterIP) &&
+		reflect.DeepEqual(new.Spec.ClusterIPs, old.Spec.ClusterIPs) &&
```

Comment:
why this list doesn't include `*.Spec.LoadBalancerIP `

Reply:
scratch that this field will be going away soon https://github.com/kubernetes/kubernetes/pull/107235